elements/bulbs-liveblog/bulbs-liveblog.js

```diff
@@ -0,0 +1,238 @@
+    this.entriesData = {};
+
+  }
+
+  bindHandlers () {
+    this.handleInterval = this.handleInterval.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleFirebaseValue = this.handleFirebaseValue.bind(this);
+    this.handleEntryAttached = this.handleEntryAttached.bind(this);
+    this.handleEntryDetached = this.handleEntryDetached.bind(this);
+  }
+
+  detachedCallback () {
+    this.teardownFirebase();
+    this.teardownInterval();
+  }
+
+  setupFirebase () {
+    let firebaseAppConfig = {
+      apiKey: this.getAttribute('firebase-api-key'),
+      databaseURL: this.getAttribute('firebase-url'),
+    };
+    let firebaseAppName = `liveblog-${this.getAttribute('liveblog-id')}`;
+
+    this.firebaseApp = firebase.initializeApp(firebaseAppConfig, firebaseAppName);
+    this.firebaseDatabase = this.firebaseApp.database();
+    this.firebaseRef = this.firebaseDatabase
+                        .ref(this.getAttribute('firebase-path'))
+                        .orderByChild('published')
+                        .limitToLast(100);
+  }
+
+  setupInterval () {
+    this.interval = setInterval(this.handleInterval, LIVEBLOG_LATENCY);
+  }
+
+  setupEvents () {
+    this.addEventListener('click', this.handleClick);
+    this.addEventListener('liveblog-entry-attached', this.handleEntryAttached);
+    this.addEventListener('liveblog-entry-detached', this.handleEntryDetached);
+    this.firebaseRef.on('value', this.handleFirebaseValue);
+  }
+
+  teardownFirebase () {
+    this.firebaseRef.off();
+  }
+
+  teardownInterval () {
+    clearInterval(this.interval);
+  }
+
+  handleFirebaseValue (snapshot) {
+    this.entriesData = snapshot.val() || {};
+    Object.keys(this.entriesData).forEach((entryId) => {
+      parseEntry(this.entriesData[entryId]);
+    });
+    this.handleInterval();
+  }
+
+  handleInterval () {
+    if (this.fetching) {
+      return;
+    }
+
+    let entryIds = this.getEntryIdsToFetch();
+    if (entryIds.length) {
+      this.handleBlogUpdate(entryIds);
+    }
+  }
+
+  handleEntryAttached () {
+    let thisEntry = {
+      element: event.target,
+      published: event.target.published,
+    };
+
+    this.entriesStore.all[event.target.getAttribute('entry-id')] = thisEntry;
+
+    if (!this.entriesStore.oldestEntryDate || thisEntry.published < this.entriesStore.oldestEntryDate) {
+      this.entriesStore.oldestEntryDate = thisEntry.published;
+    }
+  }
+
+  handleEntryDetached () {
+    delete this.entriesStore.all[this.getAttribute('entry-id')];
+  }
+
+  getEntryIdsToFetch () {
+    let now = new Date();
+    let entryIds = [];
+    Object.keys(this.entriesData).forEach((entryId) => {
+      let entry = this.entriesData[entryId];
+      if (entry.published && entry.published < now && !this.entriesStore.all[entryId]) {
+        if (this.entriesStore.oldestEntryDate) {
+          if (entry.published >= this.entriesStore.oldestEntryDate) {
+            entryIds.push(entryId);
```

Comment:
Can we tighten up the logic in these nested `if` statements? It's difficult to understand at a glance and confusing that two different branches end up in the same place.